AutoDraft/NHL_API.py

```diff
@@ -0,0 +1,499 @@
+    player = player_response.content
+    player = pd.read_json(player)
+    player = player.people[0]
+    player_name = player['fullName']
+    player_position = player['primaryPosition']['code']
+    return player_name, player_position
+
+def assemble_multiplayer_stat_dataframe(player_id_list=[8477934, 8476356, 8473468], season_id_list=[20152016, 20162017, 20172018, 20182019], stat_list=['cumStatpoints'], shape='cols'):
+    # returns game-by-game stats for given list of players across given seasons (can specify sspecific stats for smaller returns)
+    multiplayer_df = pd.DataFrame() # TODO: add a progress bar
+    for player_id in player_id_list: 
+        player_name, player_position = get_player_name_position(player_id)
+        if player_position == 'G': continue # don't include goalies
+        if len(season_id_list) == 1: # handle single or multiple season input lists
+            player_df = augment_player_dataframe(get_player_season_game_stats(player_id=player_id, season_id=season_id_list[0]))
+        else:
+            player_df = augment_player_dataframe(get_combined_player_season_game_stats(player_id=player_id, season_id_list=season_id_list))
+        if len(stat_list) != 0: # if a specific stat is given, only grab that
+            player_small_df = player_df.loc[:, ['date', 'gameNumber'] + stat_list] # keep the useful indices
+        else:
+            player_small_df = player_df # grab it all otherwise
+        player_small_df.reset_index(drop=True, inplace=True) # get rid of messy index
+        try:
+            player_small_df.insert(0, 'name', [player_name for _ in range(len(player_small_df))])
+        except ValueError: # still don't know why this is thrown sometimes
+            st.dataframe(player_small_df)
+            player_small_df.insert(0, 'errorName', [player_name for _ in range(len(player_small_df))])
+        # player_small_df.set_index('gameNumber', inplace=True)
+        # player_small_df.rename(columns={stat: player_name}, inplace=True)
+        multiplayer_df = pd.concat([multiplayer_df, player_small_df], axis=0)
+        multiplayer_df.to_csv('./data/assemble_multiplayer_stat_dataframe_TEMP.csv') # export temporary state
+    multiplayer_df.reset_index(drop=True, inplace=True)
+    if shape == 'rows': # TODO: fix this when required to feed data in a row-per-player fashion
+        multiplayer_df = multiplayer_df.transpose()
+        # multiplayer_df.set_index(player_id_list, inplace=True)
+        try:
+            multiplayer_df.insert(0, 'name', multiplayer_df.index)
+        except ValueError:
+            multiplayer_df.insert(0, 'errorName', multiplayer_df.index)
+        multiplayer_df.insert(0, 'playerId', player_id_list)
+        multiplayer_df.set_index('playerId', inplace=True)
+    return multiplayer_df
+
+def get_all_rosters(season_id_list=[20152016, 20162017, 20172018, 20182019]): # returns roster of all players across a given list of seasons
```

Comment:
Nice data collection functions. Maybe hitting the API is no problem, but would it be helpful to write the data to disk so you're guaranteed to be able to access it in the future?

Reply:
I've got the results of the functions exported to the data folder; I will make sure to include in the main README as well as the function README (is this the best way to do this?).